platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -47,8 +53,21 @@ class Chassis(ChassisBase):
     """Platform-specific Chassis class"""
 
     def __init__(self):
+        global REBOOT_CAUSE_ROOT
         super(Chassis, self).__init__()
 
+        # adaptively reboot cause root dir initialization
+        REBOOT_CAUSE_ROOT = HWMGMT_SYSTEM_ROOT
+        if not os.path.exists(REBOOT_CAUSE_ROOT):
+            log_warning("reboot cause dir {} doesn't exist, trying other alternatives".format(REBOOT_CAUSE_ROOT))
+            if os.path.exists(HWMON_ROOT):
+                REBOOT_CAUSE_ROOT = HWMON_ROOT
+            elif os.path.exists(HWMON_ROOT_ALT):
+                log_warning("reboot cause dir {} doesn't exist, trying other alternatives".format(HWMON_ROOT))
+                REBOOT_CAUSE_ROOT = HWMON_ROOT_ALT
+            else:
+                log_warning("reboot cause dir {} doesn't exist, unable to fetch reboot cause".format(HWMON_ROOT_ALT))
+
```

Comment:
according to the low level team, this handling is not stable since hwmonX can be any numbers. have to update the logic.

Reply:
fixed.